packages/ui/src/components/Dialog/Dialog.test.tsx

```diff
@@ -0,0 +1,192 @@
+
+    expect(drawer).toHaveClass('utrecht-dialog');
+  });
+
+  it('is not visible by default', () => {
+    const { container } = render(<Dialog />);
+
+    const drawer = container.querySelector(':only-child');
+
+    expect(drawer).not.toBeVisible();
+  });
+
+  it('can have a additional class name', () => {
+    const { container } = render(<Dialog className="large" />);
+
+    const drawer = container.querySelector(':only-child');
+    expect(drawer).toHaveClass('large');
+    expect(drawer).toHaveClass('utrecht-dialog');
+  });
+
+  it('renders rich text content', () => {
+    const { container } = render(
+      <Dialog>
+        <h1>Hello, world</h1>
+      </Dialog>,
+    );
+
+    const drawer = container.querySelector(':only-child');
+
+    const richText = drawer?.querySelector('h1');
+
+    expect(richText).toBeInTheDocument();
+  });
+
+  it('can be hidden', () => {
+    const { container } = render(<Dialog hidden />);
+
+    const drawer = container.querySelector(':only-child');
+
+    expect(drawer).not.toBeVisible();
+  });
+
+  it('supports ForwardRef in React', () => {
+    const ref = createRef<HTMLDialogElement>();
+
+    const { container } = render(<Dialog ref={ref} />);
+
+    const drawer = container.querySelector(':only-child');
+
+    expect(ref.current).toBe(drawer);
+  });
+
+  describe('open dialog', () => {
+    it('renders an HTML dialog element', () => {
+      const { container } = render(<Dialog open />);
+
+      const drawer = container.querySelector('dialog:only-child');
+
+      expect(drawer).toBeInTheDocument();
+    });
+
+    it('can be opened with `open` attribute', () => {
+      const { container } = render(<Dialog open />);
+      const drawer = container.querySelector(':only-child');
+
+      expect(drawer).toBeVisible();
+    });
+
+    it('can be hidden', () => {
+      const { container } = render(<Dialog open hidden />);
+
+      const drawer = container.querySelector(':only-child');
+
+      expect(drawer).not.toBeVisible();
+    });
+  });
+  describe('header', () => {
+    it('renders a header', () => {
+      const { container } = render(<Dialog header={{ title: 'Hello, world' }} />);
+
+      const header = container.querySelector('.utrecht-dialog__header');
+
+      expect(header).toBeInTheDocument();
+    });
+    it('renders a heading 1 by default', () => {
+      const { container } = render(<Dialog header={{ title: 'Hello, world' }} />);
+
+      const heading = container.querySelector('h1');
+
+      expect(heading).toBeInTheDocument();
+    });
+    it('renders a custom heading level', () => {
+      const { container } = render(<Dialog header={{ title: 'Hello, world', headingLevel: 2 }} />);
+
+      const heading = container.querySelector('h2');
+
+      expect(heading).toBeInTheDocument();
+    });
+    it('renders a close button by default', () => {
+      const { container } = render(<Dialog header={{ title: 'Hello, world', closeButton: {} }} />);
```

Comment:
Ik stel voor om te beginnen met meer atomic components en meer flexibiliteit, en dan later kijken op welke manier mensen `Dialog` het meest gebruiken. Dan kunnen we daar op de properties kiezen.

Bijvoorbeeld:

```tsx
<Dialog>
  <DialogHeader>
    <HTMLHeading level={2}>Dialog heading</HTMLHeading>
    <DialogCloseButton />
  </DialogHeader>
  <DialogBody>
    ...
  </DialogBody>
  <DialogFooter>
    ...
  </DialogFooter>
</Dialog>
```

We kunnen natuurlijk allebei hebben! We kunnen een `StrapiDialog` ofzo maken die wel al die properties heeft voor onze eigen convenience. Maar de component die later naar het design system moet heeft meer flexibiliteit nodig.